### ISSUE TITLE
ci: Update CentOS 7 in stage: build-and-test-in-container

### ISSUE THREAD

**Issue by aleksraiden** [apache/kvrocks#2101]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/kvrocks/issues) and found no similar issues.

### Motivation

At summer 2024 we have a two deprecates - Github runner-s. based on NodeJS 16 and CentOS 7 (see: https://blog.centos.org/2023/04/end-dates-are-coming-for-centos-stream-8-and-centos-linux-7/)

### Solution

So, we can't continuous use old runners and need to switch from CentOS 7 to other OS where we can use NodeJS 20.  

Lets talk for best solution

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

**Comment by aleksraiden** [comment]
In addition, an docker image not supported too - https://hub.docker.com/_/centos 

> This image is no longer supported/maintained (non-EOL tags last updated November 16, 2020

So, maybe we switch into Ubuntu?

**Comment by aleksraiden** [comment]
@tisonkun @PragmaTwice @git-hulk

**Comment by jihuayu** [comment]
I think we can ignore the centos7 warning.

At least in China, a large number of enterprises are still using centos7, it is necessary to keep the CI of centos7.

I've researched this issue and the reason centos7 can't use nodejs20 is because the libc version of nodejs20 is incompatible with the version that comes with centos7. However, kvrocks also depends on libc, so we can't upgrade the libc version for nodejs20. 
To summarize, there is no way to solve this problem in my opinion.

So I suggest keeping this annoying warning

**Comment by aleksraiden** [comment]
@jihuayu lot of thanks about research. But, as I see, when Github are moved all runners to NodeJS 20 (see https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/) our pipelines will not work

**Comment by aleksraiden** [comment]
So, what we can do next? From yesterday, CentOS repo has no working. Some a archive are there - http://archive.kernel.org/centos-vault/

Now all CI pipelines are broken at this stage

**Comment by git-hulk** [comment]
I'm in favor of removing support of CentOS7 in CI due to it has been EOL.
